### PR TITLE
Fix: Global styles for non-core blocks aren't loaded on the front end

### LINKED ISSUE #43679: Custom Block, theme.json definitions not loaded with latest RC Gutenberg Plugin
### Description

Hi,
I'm just trying the latest Gutenberg RC and I can see that something is not working anymore.
The issue is that I created a custom excerpt block, called theme/single-post-excerpt and I added some css to the theme.json:

`			"theme/single-post-excerpt": {
				"typography": {
					"lineHeight": "var(--wp--custom--typography--line-height--small)",
					"fontSize": "var(--wp--preset--font-size--large)",
					"fontWeight": "500"
				}
			}`

That definitions works good on the latest WP 6.0.1 but if I install the latest Gutenberg RC they are ignored.

### Step-by-step reproduction instructions

Create a custom block.
Add css definitions for that block on the theme.json
Check if they are applied.

### Screenshots, screen recording, code snippet

_No response_

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by t-hamano**
Fix #43679

## What?
This PR fixes that global styles for non-core blocks aren't loaded on the front end,

## Why?
I think this is due to the fact that non-core blocks were excluded in [wp_add_global_styles_for_blocks](https://github.com/WordPress/gutenberg/blob/7f0a211da6603b336b75b75d8235a50bf547e6f3/lib/compat/wordpress-6.1/get-global-styles-and-settings.php#L23).

## How?
I have applied [the logic of script handle generation in the WordPress core.](https://github.com/WordPress/wordpress-develop/blob/438822aa5b8eab673c4b372474d6cb69bd9799da/src/wp-includes/script-loader.php#L2593-L2609)
This will ensure that global styles for non-core blocks and global styles for the elements property of non-core blocks are output correctly.

## Testing Instructions
- Checkout this branch and build gutenberg.
- run `npx @wordpress/create-block gutenpride` outside gutenberg repository.
- Define the following in `.wp-env.override.json`.

```javascript

{
	"plugins": [ ".", "../path/to/gutenpride" ],
}

```

- run `npm run wp-env start -- --update`
- Activate Gutenpride plugin.
- Activate the `emptytheme` and put the following code in `theme.json`.

```json
{
	"$schema": "https://schemas.wp.org/trunk/theme.json",
	"version": 2,
	"styles": {
		"blocks": {
			"create-block/gutenpride": {
				"border": {
					"width": "3px",
					"color": "#f00",
					"style": "solid"
				},
				"color": {
					"text": "#fff",
					"background": "#000"
				},
				"elements": {
					"button": {
						"color": {
							"text": "#fff"
						}
					}
				}
			}
		}
	}
}
```
- Open the post editor.
- Insert the Gutenpride block and confirm that the style is applied.

### Global styles should be output via the block handle on the front-end

- Confirm that the global style is generated inside the block handle (`create-block-gutenpride-style-inline-css`) on the front end.

handle

### Global styles should be output via the global-styles handle on the front-end

- Add `add_filter( 'should_load_separate_core_block_assets', '__return_false' );` in `functions.php`. 
- Confirm that the global style is generated inside the global handle (`global-styles-inline-css`) on the front end.

global

### CODE REVIEW

File: lib/compat/wordpress-6.1/get-global-styles-and-settings.php [hunk 1]
```diff
@@ -33,16 +45,23 @@ function wp_add_global_styles_for_blocks() {
 				array_filter(
 					$metadata['path'],
 					function ( $item ) {
-						if ( str_contains( $item, 'core/' ) ) {
+						if ( str_contains( $item, '/' ) ) {
```
Comment (t-hamano) [inline lib/compat/wordpress-6.1/get-global-styles-and-settings.php:48]: I think to check only slashes is the only way to include non-core blocks, is there a better approach?

**Comment by t-hamano** [comment]
The GitHub action failing will be resolved  if #44052 is merged.

**Comment by scruffian** [comment]
I did something very similar here! https://github.com/WordPress/gutenberg/pull/44018

**Comment by t-hamano** [comment]
Sorry, I should have checked my past PRs.
I would like to close this PR.

**Pull request closed without merging**